cabal-testsuite/PackageTests/NewBuild/CmdTest/OptionsFlag/cabal.test.hs

```diff
@@ -0,0 +1,12 @@
+import Test.Cabal.Prelude
+
+main = cabalTest $ do
+    res <- cabal' "v2-test"
+             [ "--test-show-details=direct"
+
+             , "--test-option=1"
+             , "--test-options=\"2 3\""
+             , "--test-option=4"
+             , "--test-options=\"5 6\""
+             ]
+    assertOutputContains "1 2 3 4 5 6" res
```

Comment:
I think you should use the same strategy as the Setup test I linked to, namely let the test executable check the arguments and exit accordingly. That way we don't have to rely on matching on the output. I also don't like that you have to use `--test-show-details=direct` here but that should be irrelevant if you let the executable do the check.

Reply:
I've changed both test cases to check the output in the executable.